pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -212,112 +207,110 @@ func (p *DefaultProvider) ensureLaunchTemplate(ctx context.Context, options *ami
-				// with a single EFA network interface, and we should support those use cases. Launch failures with multiple enis should be considered user misconfiguration.
+		return lo.Times(options.EFACount, func(i int) ec2types.LaunchTemplateInstanceNetworkInterfaceSpecificationRequest {
+			return ec2types.LaunchTemplateInstanceNetworkInterfaceSpecificationRequest{
+				NetworkCardIndex:         lo.ToPtr(int32(i)),
+				DeviceIndex:              lo.ToPtr(lo.Ternary[int32](i == 0, 0, 1)),
+				InterfaceType:            lo.ToPtr(string(ec2types.NetworkInterfaceTypeEfa)),
+				Groups:                   lo.Map(options.SecurityGroups, func(s v1.SecurityGroup, _ int) string { return s.ID }),
 				AssociatePublicIpAddress: options.AssociatePublicIPAddress,
 			}
 		})
 	}
 
 	if options.AssociatePublicIPAddress != nil {
-		return []*ec2.LaunchTemplateInstanceNetworkInterfaceSpecificationRequest{
+		return []ec2types.LaunchTemplateInstanceNetworkInterfaceSpecificationRequest{
 			{
 				AssociatePublicIpAddress: options.AssociatePublicIPAddress,
-				DeviceIndex:              aws.Int64(0),
-				Groups:                   lo.Map(options.SecurityGroups, func(s v1.SecurityGroup, _ int) *string { return aws.String(s.ID) }),
+				DeviceIndex:              aws.Int32(0),
+				Groups: lo.Map(options.SecurityGroups, func(s v1.SecurityGroup, _ int) string {
+					return s.ID
+				}),
 			},
 		}
 	}
 	return nil
 }
 
-func (p *DefaultProvider) blockDeviceMappings(blockDeviceMappings []*v1.BlockDeviceMapping) []*ec2.LaunchTemplateBlockDeviceMappingRequest {
+func (p *DefaultProvider) blockDeviceMappings(blockDeviceMappings []*v1.BlockDeviceMapping) []*ec2types.LaunchTemplateBlockDeviceMappingRequest {
 	if len(blockDeviceMappings) == 0 {
 		// The EC2 API fails with empty slices and expects nil.
 		return nil
 	}
-	var blockDeviceMappingsRequest []*ec2.LaunchTemplateBlockDeviceMappingRequest
+	var blockDeviceMappingsRequest []*ec2types.LaunchTemplateBlockDeviceMappingRequest
 	for _, blockDeviceMapping := range blockDeviceMappings {
-		blockDeviceMappingsRequest = append(blockDeviceMappingsRequest, &ec2.LaunchTemplateBlockDeviceMappingRequest{
+		blockDeviceMappingsRequest = append(blockDeviceMappingsRequest, &ec2types.LaunchTemplateBlockDeviceMappingRequest{
 			DeviceName: blockDeviceMapping.DeviceName,
-			Ebs: &ec2.LaunchTemplateEbsBlockDeviceRequest{
+			Ebs: &ec2types.LaunchTemplateEbsBlockDeviceRequest{
 				DeleteOnTermination: blockDeviceMapping.EBS.DeleteOnTermination,
 				Encrypted:           blockDeviceMapping.EBS.Encrypted,
-				VolumeType:          blockDeviceMapping.EBS.VolumeType,
-				Iops:                blockDeviceMapping.EBS.IOPS,
-				Throughput:          blockDeviceMapping.EBS.Throughput,
+				VolumeType:          ec2types.VolumeType(aws.ToString(blockDeviceMapping.EBS.VolumeType)),
+				Iops:                convertToNil(blockDeviceMapping.EBS.IOPS),
```

Comment:
I created this separately because unfortunately unless we update IOPs and Throughputs types to int32 instead of int64 and bump the hash they will need to be cast to int32 but that changes nil to 0 which breaks elsewhere. I will handle it in a separate PR.

Reply:
I think you can still use here `lo.ToPtrOr()` https://github.com/samber/lo?tab=readme-ov-file#fromptror

Reply:
ToPtrOr does not exist only FromPtrOr

Reply:
https://github.com/samber/lo?tab=readme-ov-file#emptyabletoptr